### PR TITLE
Allows the use of shovel on stacks of grass

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Adds the exact same mechanic as shoveling grass turfs to shoveling actual grass tile stacks, time to perform this action depends on amount of grass tiles in a stack.

Did not add it to basalt tiles, as such situation never comes up.

## Why It's Good For The Game

Instead of using hotkeys spamming hundreds of clicks/buttons to perform this action, the game now allows directly using a shovel on a stack of grass tiles, greatly reducing hand pain while keeping the time required similar.

## Changelog
:cl:
add: You can now use a shovel directly on stacks of grass tiles.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
but why does grass turn into sand?

**Comment by ItsTisme** [comment]
> but why does grass turn into sand?

bluespace grass

Also my keyboard thanks you for this gift

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -76,6 +76,17 @@
 	turf_type = /turf/open/floor/grass
 	resistance_flags = FLAMMABLE
 
+/obj/item/stack/tile/grass/attackby(obj/item/W, mob/user, params)
+	if((W.tool_behaviour == TOOL_SHOVEL) && params)
+		to_chat(user, "<span class='notice'>You start digging up [src]</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/stacks/tiles/tile_types.dm]:
```suggestion
		to_chat(user, "<span class='notice'>You start digging up [src].</span>")
```

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 2]
```diff
@@ -76,6 +76,17 @@
 	turf_type = /turf/open/floor/grass
 	resistance_flags = FLAMMABLE
 
+/obj/item/stack/tile/grass/attackby(obj/item/W, mob/user, params)
+	if((W.tool_behaviour == TOOL_SHOVEL) && params)
```
Comment (PowerfulBacon) [inline code/game/objects/items/stacks/tiles/tile_types.dm:80]: why && params?

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 3]
```diff
@@ -76,6 +76,17 @@
...
+		to_chat(user, "<span class='notice'>You start digging up [src]</span>")
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, 1)
+		if(do_after(user, 2 * get_amount(), target = src))
+			new /obj/item/stack/ore/glass(get_turf(src), 2 * get_amount())
+			user.visible_message("[user] digs up [src].", "<span class='notice'>You uproot [src].</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/stacks/tiles/tile_types.dm]:
```suggestion
			user.visible_message("<span class='notice'>[user] digs up [src].</span>", "<span class='notice'>You uproot [src].</span>")
```

**Comment by eeSPee** [comment]
Not sure if intentional, but now you can plant grass >>> shovel >>> get sand >>> make soil.

**Comment by ghost** [comment]
> but why does grass turn into sand?

It always turned into sand ever since grass existed as a floor type. What this does is streamline the process without requiring placing of the grass tiles upon a floor and then shoveling it, one grass tile at a time.


> Not sure if intentional, but now you can plant grass >>> shovel >>> get sand >>> make soil.

That is exactly what grass in botany is used for, and how it was used for on every codebase that has grass, both bay and tg derivatives use it as far as I remember.

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/items/stacks/tiles/tile_types.dm:80]: I have copied the exact same behavior from how a shovel is applied to a grass turf, I do not know why it was used there.

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 3 above)
Comment (ghost) [reply code/game/objects/items/stacks/tiles/tile_types.dm]: Again, this is a direct copy of grass turf behavior.

**Comment by eeSPee** [comment]
Huh. That's a good change then.

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/stacks/tiles/tile_types.dm]: Again, a direct copy of grass turf behavior.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**